game/scripts/vscripts/components/duels/duels.lua

```diff
@@ -89,278 +44,106 @@ function Duels:StartDuel ()
-    DebugPrint('Checking player number ' .. playerNumber)
-    local goodGuy = Duels:GetUnassignedPlayer(goodPlayers, goodPlayerIndex)
-    local badGuy = Duels:GetUnassignedPlayer(badPlayers, badPlayerIndex)
-    local goodPlayer = PlayerResource:GetPlayer(goodGuy.id)
-    local badPlayer = PlayerResource:GetPlayer(badGuy.id)
-    local goodHero = goodPlayer:GetAssignedHero()
-    local badHero = badPlayer:GetAssignedHero()
-
-    goodGuy.duelNumber = 2
-    badGuy.duelNumber = 2
-
-    FindClearSpaceForUnit(goodHero, spawn1, true)
-    FindClearSpaceForUnit(badHero, spawn2, true)
-
-    Duels.zone2.addPlayer(goodGuy.id)
-    Duels.zone2.addPlayer(badGuy.id)
+  local playerSplitOffset = math.random(0, maxPlayers)
 
-    Duels:MoveCameraToPlayer(goodGuy.id, goodHero)
-    Duels:MoveCameraToPlayer(badGuy.id, badHero)
+  Duels:ShuffleTeams()
 
-    -- disable respawn
-    goodHero:SetRespawnsDisabled(true)
-    badHero:SetRespawnsDisabled(true)
+  local duelists = {{},{}}
+  for playerNumber = 1,playerSplitOffset do
+    for index, team in pairs(self.teams) do
+	    table.insert(duelists[index], team[playerNumber])
+	  end
   end
 
-  Duels.currentDuel = {
-    goodLiving1 = playerSplitOffset,
-    badLiving1 = playerSplitOffset,
-    goodLiving2 = maxPlayers - playerSplitOffset,
-    badLiving2 = maxPlayers - playerSplitOffset,
-    duelEnd1 = playerSplitOffset == 0,
-    duelEnd2 = maxPlayers == playerSplitOffset,
-    badPlayers = badPlayers,
-    goodPlayers = goodPlayers,
-    badPlayerIndex = badPlayerIndex,
-    goodPlayerIndex = goodPlayerIndex
-  }
-
-  Timers:CreateTimer(90, Dynamic_Wrap(Duels, 'EndDuel'))
-end
+  local spawns = Duels:GetSpawns(1)
+  self.duel1 = duelLib.Duel(duelists, spawns, self.zone1)
 
-function Duels:MoveCameraToPlayer (playerId, entity)
-  PlayerResource:SetCameraTarget(playerId, entity)
-
-  Timers:CreateTimer(1, function ()
-    PlayerResource:SetCameraTarget(playerId, nil)
-  end)
-end
-
-function Duels:GetUnassignedPlayer (group, max)
-  while true do
-    local playerIndex = math.random(1, max)
-    if group[playerIndex].assigned == nil then
-      group[playerIndex].assigned = true
-      return group[playerIndex]
-    end
+  local duelists2 = {{},{}}
+  for playerNumber = playerSplitOffset+1, maxPlayers do
+    for index, team in pairs(self.teams) do
+	    table.insert(duelists[index], team[playerNumber])
+	  end
   end
-end
 
-function Duels:EndDuel ()
-  if Duels.currentDuel == nil then
-    DebugPrint ('There is no duel running')
-    return
-  end
+  local spawns2 = Duels.GetSpawns(2)
+  self.duel2 = duelLib.Duel(duelists2, spawns2, self.zone2)
 
-  DebugPrint('Duel has ended')
+  Timers:CreateTimer(90, Dynamic_Wrap(Duels, 'EndDuels'))
+end
 
+function Duels:EndDuels ()
   local nextDuelIn = 300
   -- why dont these run?
-  Timers:CreateTimer(nextDuelIn, Dynamic_Wrap(Duels, 'StartDuel'))
+  Timers:CreateTimer(nextDuelIn, Dynamic_Wrap(Duels, 'Warn'))
```

Comment:
Yep, changed it to something more descriptive in the browser and didn't get all the references, woops